### PR TITLE
Fix duplicate cookie expiration calls in the CookieJar implementation

### PR DESCRIPTION
**Pull request by bdraco**
## What do these changes do?

`_do_expiration` was already being called `filter_cookies` because it iterates `self` which calls `__iter__` which also calls `_do_expiration`

Calling `len()` on the cookie jar has the side of expiring stale cookies since it also called `__iter__`

## Are there changes in behavior for the user?

Calling `len()` on the cookie jar no longer has the side effect of expiring stale cookies.

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by bdraco** [comment]
Screenshot 2023-11-02 at 10 19 01 PM

**Comment by bdraco** [comment]
looks like there is another call
Screenshot 2023-11-02 at 11 01 02 PM

**Comment by bdraco** [comment]
Screenshot 2023-11-02 at 11 17 21 PM


Only called once now per filter_cookies

**Comment by bdraco** [comment]
It also looks like every time a request is made without a hostname it has to reconstruct a netloc to get `origin` and make an ip address object
Screenshot 2023-11-02 at 11 19 03 PM

**Comment by bdraco** [comment]
confirmed only happening once per filter_cookies now

filter_cookies_one_parent

File: aiohttp/cookiejar.py [hunk 1]
```diff
@@ -234,7 +239,6 @@ def filter_cookies(
         self, request_url: URL = URL()
     ) -> Union["BaseCookie[str]", "SimpleCookie[str]"]:
         """Returns this jar's cookies filtered by their attributes."""
-        self._do_expiration()
```
Comment (Dreamsorcerer) [inline aiohttp/cookiejar.py]: I think this'll conflict with #7777, so I'd suggest leaving it in for now. Then that PR should end up removing the iter over self.

**Comment by bdraco** [comment]
Marking as draft until #7777 merges as it will be much easier to rebase this PR vs that one

File: aiohttp/cookiejar.py (hunk 1 above)
Comment (Dreamsorcerer) [reply aiohttp/cookiejar.py]: This PR should be unblocked now.

**Review by Dreamsorcerer — APPROVED** [review]

**Pull request merged**